src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Packaging/ProvideDiffSupportedContentTypeAttribute.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio
+{
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
+    internal class ProvideDiffSupportedContentTypeAttribute : RegistrationAttribute
+    {
+        private const string RegistryKey = @"Diff\SupportedContentTypes";
+
+        private readonly string _contentType;
+        private readonly string _extension;
+
+        public ProvideDiffSupportedContentTypeAttribute(string extension, string contentType)
```

Comment:
ContextType is never used, just remove it and name the attribute after what "empty" means.

Reply:
ContentType is used, its the value of the registry key

Reply:
It's always empty, this doesn't need to be a parameter.